src/DataGrid.tsx

```diff
@@ -191,7 +191,7 @@ function DataGrid<R, K extends keyof R>({
     });
   }, [columnMetrics, scrollLeft]);
 
-  const viewportColumns: readonly CalculatedColumn<R>[] = useMemo(() => {
+  const viewportColumns: readonly CalculatedColumn<R, never>[] = useMemo(() => {
```

Comment:
```suggestion
  const viewportColumns: readonly CalculatedColumn<R, SR>[] = useMemo(() => {
```

Reply:
For regular `Row` related definitions, what's the point for passing the `SR` into them? Same question for the `Column` definition being used in calculation.

Reply:
>For regular Row related definitions, what's the point for passing the SR into them?

Because it's the correct thing to do.
We pass these columns back to user code, so their types much match the types the user code has, even if it's outside the summary row rendering context..

Reply:
It's still not convincing to me. For the regular `Row`, in the `Row` context, it doesn't have the `Summary Row` definition at all which will always point to `never`. That's why I thought it make sense to default `SummaryRow` to `never`.

http://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-3.html#generic-parameter-defaults

My understanding for the default type is that, the user doesn't need to constantly put the default type whey they use it. In our case, most of time when the grid doesn't have summary row being defined or passed, it is just `never`, so the user doesn't need to explicitly write like 
```
return <DataGrid<Row, 'id', never> xxx />;
```
and the following would just work
```
return <DataGrid<Row, 'id'> xxx />;
```

Reply:
Imagine this:
```
<DataGrid
  onSomeEvent={({ column }) => myEventHandler(column)}
/>
```
Here your `myEventHandler` will use the `Column<Row, SummaryRow>` type, so `Column<Row, never>` will be incompatible.

Reply:
In your example, I think it makes sense to define `myEventHandler` as `myEventHandler(c: Column<Row, never>) {}`, and the call back functions should follow RDG type definition. However, I will try to remove all `= never` in another commit.